teleirc.js

```diff
@@ -93,226 +22,6 @@ let ircbot = new irc.Client(config.irc.server, config.irc.botName, {
-                }
-            } else {
-                console.log("Ignoring non-text message: " + JSON.stringify(msg));
-            }
-        } else {
-            // Relay all text messages into IRC
-            ircbot.say(config.irc.channel, config.irc.prefix + from + config.irc.suffix + " " + message);
-        }
-    } else {
-        // Messages that are sent to the bot outside of the group chat should just be dumped
-        // to the console for potential testing and debugging to do things like check chat IDs
-        // and verify the JSON formats of various messages
-        console.log("[TG Debug] " + JSON.stringify(msg));
-    }
-});
-
-// Action to invoke on incoming messages from the IRC side
-ircbot.addListener('message', (from, channel, message) => {
-    let matchedNames = config.ircBlacklist.filter(function (name) {
-        return from.toLowerCase() === name.toLowerCase();
-    });
-
-    if (matchedNames.length <= 0) {
-        sendTelegramMessage(config.tg.chatId, from + ": " + message);
-    }
-
-});
-
-ircbot.addListener('error', (message) => {
-    console.log("[IRC Debug] " + JSON.stringify(message));
-});
-
-// These additional alerts can be turned on in the config file
-if (config.tg.showActionMessage) {
-    ircbot.addListener('action', (from, channel, message) => {
-        let matchedNames = config.ircBlacklist.filter(function (name) {
-            return from.toLowerCase() === name.toLowerCase();
-        });
-
-        if (matchedNames.length <= 0) {
-            sendTelegramMessage(config.tg.chatId, from + " " + message);
-        }
-    });
-}
-
-if (config.tg.showJoinMessage) {
-    // Let the telegram chat know when a user joins the IRC channel
-    ircbot.addListener('join', (channel, username) => {
-        sendTelegramMessage(config.tg.chatId, username + " has joined " + channel + " channel.");
-    });
-}
-
-if (config.tg.showLeaveMessage) {
-    // Let the telegram chat know when a user leaves the IRC channel
-    ircbot.addListener('part', (channel, username, reason) => {
-        if (typeof reason != "string") {
-            reason = "Parting...";
-        }
-        sendTelegramMessage(config.tg.chatId, username + " has left " + channel + ": " + reason + ".");
-    });
-}
-
-if (config.tg.showKickMessage) {
-    // Let the telegram chat know when a user is kicked from the IRC channel
-    ircbot.addListener('kick', (channel, username, by, reason) => {
-        if (typeof reason != "string") {
-            reason = "Kicked";
-        }
-        sendTelegramMessage(config.tg.chatId, username + " was kicked by " + by + " from " + channel + ": " + reason + ".");
-    });
-}
-
-// Quick function to print an error message and bail out if the
-// config is missing any required settings in the config file.
-function setupError(message) {
-    console.warn("[setup] " + message);
-    process.exit(1);
-}
+teleIrc.initStage3_initBots(ircbot, tgbot);
+teleIrc.initStage4_addIrcListeners();
```

Comment:
Adding some additional logging between each step; not within the class, but out here, might be worthwhile for debugging purposes. It would make it easier to tell at which stage an issue came up if the bot failed to start.